src/utils/WeakMap.js

```diff
@@ -0,0 +1,9 @@
+class NonTrackingDummyWeakMap {
+  get() {}
+  set() {}
+  has() {
+    return false
+  }
+}
+
+export default WeakMap || NonTrackingDummyWeakMap
```

Comment:
I think this would throw a `ReferenceError` if `WeakMap` is not defined.

Maybe we could go with something like the following:

```js
export default (typeof WeakMap === 'undefined'
  ? NonTrackingDummyWeakMap
  : WeakMap)
```

Reply:
@matthieuprat ok updated